feat: add-README for rollups-validator-node chart (#23)
cartesi/helm-charts: charts/rollups-validator-node/README.md

# rollups-validator-node

# Package for Cartesi Rollups validator nodes

![Version: 0.5.1](https://img.shields.io/badge/Version-0.5.1-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 0.9.1](https://img.shields.io/badge/AppVersion-0.9.1-informational?style=flat-square)

Refer to the [official documentation](https://docs.cartesi.io/cartesi-rollups/overview/) for more information about Cartesi Rollups.

## Maintainers

| Name | Email | Url |
| ---- | ------ | --- |
| endersonmaia |  |  |
| oap75 |  |  |

> 

Reviewer: This is generated via template, maybe we need to improve the chart.yaml to add better maintainers information.

<!--- app-name: Redis&reg; -->

> # Cartesi package for rollups validator nodes

Reviewer: We could even rename this chart in the future to `cartesi/rollups-node`.

> [Cartesi rollups](https://docs.cartesi.io/cartesi-rollups/overview/) is an open source, a set of on-chain and off-chain components that implement an Optimistic Rollups solution and provide the general framework for building DApps.

Reviewer 2: Should we really add this definition here? Perhaps we could simply refer the reader to the official docs:

```suggestion
Refer to the [official documetation](https://docs.cartesi.io/cartesi-rollups/overview/) for more information about Cartesi Rollups.
```

> [Cartesi rollups](https://docs.cartesi.io/cartesi-rollups/overview/) is an open source, a set of on-chain and off-chain components that implement an Optimistic Rollups solution and provide the general framework for building DApps.

Reviewer 2: I'd also suggest replacing `rollups` with `Rollups` throughout the document.

> [Cartesi rollups](https://docs.cartesi.io/cartesi-rollups/overview/) is an open source, a set of on-chain and off-chain components that implement an Optimistic Rollups solution and provide the general framework for building DApps.

Author: Fixed !

## TL;DR

```console

> helm install my-release oci://registry-1.docker.io/cartesi/rollups-validator-node

Reviewer: and I think we could remove the `registry-1.` part.

> helm install my-release oci://registry-1.docker.io/cartesi/rollups-validator-node

Reviewer: maybe is also relevant for us to inform about the option without using `oci://`, what do you think?

> helm install my-release oci://registry-1.docker.io/cartesi/rollups-validator-node

Author: Yes, we can add another OR section to use the old way

```

### OR

```console
helm install my-release oci://ghcr.io/cartesi/charts/rollups-validator-node
```

## Introduction

This chart bootstraps [Cartesi Rollups Validator Nodes](https://docs.cartesi.io/cartesi-rollups/overview/) deployments on a [Kubernetes](https://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.

## Prerequisites

- Kubernetes 1.19+
- Helm 3.2.0+

> - PV provisioner support in the underlying infrastructure

Reviewer: ```suggestion
- PersistentVolume provisioner support in the underlying infrastructure
```

Author: Fixed !

| --------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------ |
| `postgresql.enabled`                            | Use bitnami postgreSQL pod.                                                                                                 | `true`                  |
| `postgresql.auth`                     | Set bitnami postgreSQL`username`, `password`,`database`                                                                                                            | `""`              |
| `postgreSQL.image.tag`                            | bitnami postgreSQL docker image tag.                                                                                                 | `13.9.0-debian-11-r27`                  |

Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,

```console
helm install my-release \
  --set postgresql.auth.password=secretpassword \

>     oci://registry-1.docker.io/cartesi/rollups-validator-node

Author: This is just a template . should we specify a specific version in the template ? 

REF : https://github.com/bitnami/charts/blob/main/bitnami/redis/README.md#tldr

>     oci://registry-1.docker.io/cartesi/rollups-validator-node

Author: When we don't specify, the latest version will pull.

```

> The above command sets the bitnami postgreSQL server password to `secretpassword`.

Reviewer: ```suggestion
The above command sets the PostgreSQL server password to `secretpassword`.
```

Author: Fixed !

> NOTE: Once this chart is deployed, it is not possible to change the application's access credentials, such as usernames or passwords, using Helm. To change these application credentials after deployment, delete any persistent volumes (PVs) used by the chart and re-deploy it, or use the application's built-in administrative tools if available.

Alternatively, a YAML file that specifies the values for the parameters can be provided while installing the chart. For example,

```console

> helm install my-release -f values.yaml oci://registry-1.docker.io/cartesi/rollups-validator-node

Author: True !

> helm install my-release -f values.yaml oci://registry-1.docker.io/cartesi/rollups-validator-node

Author: Fixed !

```

> **Tip**: You can use the default [values.yaml](values.yaml)

## Configuration and installation details

### [Rolling VS Immutable tags](https://docs.bitnami.com/containers/how-to/understand-rolling-tags-containers/)

It is strongly recommended to use immutable tags in a production environment. This ensures your deployment does not change automatically if the same tag is updated with a different image.

Cartesi will release a new chart updating its containers if a new version of the main container, significant changes, or critical vulnerabilities exist.

## Persistence

> By default, the chart uses storageclass named `standard` for the [Persistent Volume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/). The volume is created using dynamic volume provisioning. If a your default storageclass name is different, specify it during installation.

Author: Fixed !

<!--- app-name: Redis&reg; -->

# Package for Cartesi Rollups validator nodes

Refer to the [official documentation](https://docs.cartesi.io/cartesi-rollups/overview/) for more information about Cartesi Rollups.

## TL;DR

```console
helm install my-release oci://docker.io/cartesi/rollups-validator-node
```

### OR

```console
helm install my-release oci://ghcr.io/cartesi/charts/rollups-validator-node
```

### OR

```console
helm repo add cartesi https://cartesi.github.io/helm-charts

> helm install my-release cartesi

Reviewer: I think we need the chart name here:

```suggestion
helm install my-release cartesi/rollups-validator-node
```

Or, instead of using `my-release`, use `my-rollups-validator-node` like we do [here](https://cartesi.github.io/helm-charts/), like that:

```suggestion
helm install my-rollups-validator-node cartesi/rollups-validator-node
```
